### ISSUE TITLE
Illegal use of conditional component in Thermal.HeatTransfer.Examples.ControlledTemperature

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1073]
**Reported by jmattsson on 10 Apr 2013 14:50 UTC**
`Modelica.Thermal.HeatTransfer.Examples.ControlledTemperature` contains the declaration:
```mo
output Modelica.SIunits.Temperature TRes(displayUnit="degC") = heatingResistor.heatPort.T;
```

However, heatingResistor.heatPort is a conditional component. Using `T_heatPort` instead of `heatPort.T` should work.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1073

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 19 Apr 2013 07:35 UTC**
Fixed in b6dda219ac092466faf9f7ad5d13c502e14aa528, as suggested above.

**Comment by modelica-trac-importer** [comment]
**Changelog modified by otter on 19 Apr 2013 07:35 UTC**
Illegal use of conditional component in Thermal.HeatTransfer.Examples.ControlledTemperature correct by replacing the reference with the correct non-conditional variable reference

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 19 Apr 2013 07:37 UTC**

**Comment by modelica-trac-importer** [comment]
**Changelog modified by otter on 19 Apr 2013 07:37 UTC**
Illegal use of conditional component in Thermal.HeatTransfer.Examples.ControlledTemperature corrected by replacing it with a non-conditional variable reference

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 19 Apr 2013 08:59 UTC**
The previous fix introduced an error, since the array portsData had no defined
dimension if use_portsData = false. This has been fixed in 7f0b63f594440907512ff9bb1412f173de852c01.
